subsys/net/l2/ppp/ppp_l2.c

```diff
@@ -341,9 +341,9 @@ static void echo_reply_handler(void *user_data, size_t user_data_len)
 {
 	struct ppp_context *ctx = user_data;
 	u32_t end_time = k_cycle_get_32();
```

Comment:
Can't k_cycle_get_32() wrap-around (e.g. on an ARM Cortex M), so ctx->shell.echo_req_data could actually be larger than end_time?

Reply:
Sure, but we are actually interested in the diff which will be correctly set even if the time has wrapped.

Reply:
ok, thx for the explanation